isc_libdemo.c

```diff
@@ -61,58 +65,267 @@ int hello_world(struct m0_buf *in, struct m0_buf *out,
 		 */
 		out_str = m0_strdup("world");
 		if (out_str != NULL)
-			m0_buf_init(out, (void *)out_str, strlen(out_str));
+			m0_buf_init(out, out_str, strlen(out_str));
 		else
 			*rc = -ENOMEM;
```

Comment:
*rc = M0_ERR(-ENOMEM), similarly in other places.

Reply:
m0tracing is not working for loadable .so modules and this won't be seen in m0traces anyway, so I don't see very much point in this.

Reply:
Well, hopefully, we fix this problem, so it will be usable then.

Reply:
We might need some insights from @chumakd about this.